cilium/CFP-39876-clustermesh-filtered-export.md

```diff
@@ -0,0 +1,156 @@
+```
+
+This explicitly marks all unannotated namespaces as global and marks the annotated services as local. Note that if both global and local annotations are present on different namespaces, the behaviour is to mark all unannotated namespaces as local(The global annotation has higher priority). Hence, to explicitly mark the unannotated services as global, customers need to ensure that no namespace is explictly marked global in their cluster. Logically, since marking a namespace as global marks all other namespaces as local, customers should not be using both the annotations for any cluster. To summarize behavior with deny annotation
+
+- A full synchronization is triggered 
+- Only entries associated with namespaces explicitly annotated with the value `false` will be skipped from the export provided no other namespace has the global annotation.
+- Entries will be added in etcd for unannotated namespaces provided no other namespaces has the global annotation.
+- All unannotated namespaces are exported unless no other namespace has the global annotation. 
+
+When the annotation is removed or disabled for a namespace:
+- A full synchronization is triggered.
+- The system exports the CiliumEndpoints and CiliumIdentities with the updated list of global namespaces
+- If no namespaces remain annotated post removal of the annotations, the behavior defaults to the existing behavior of the `clustermesh-apiserver`.
+
+#### Network Policies 
+An important part of this implementation is the functioning of network policies. In tunnel mode, the identitiy information is embedded in the network packet itself whereas in native routing mode, the identity information at the destination is derieved through the destination cluster's ipcache. Due to this descrepancy, if an identity is not exported and has associated network policies, the policies will be enforced in tunnel mode whereas they will not be honored in native mode if the identity is not exported since the identity is inferred as world at the destination due to the entry missing in the ipacache. As a result, for the network policies to work as expected, in the scoped export mode, network policies would only be honored for identities and endpoints created under `global-namespaces`.  Theoretically, the network policy descrepeancy would only occur for native routing mode but we intend to provide a consistent product expreience and communicate non support for all routing modes in non global(local) namespaces. We can provide updated documentation providing destination identity values for different routing modes but the network policy enforcement is only supported for workloads under scoped-export namespaces. 
+Similarly for Global Service implementation, since only the clients under global namespaces are enabled for cross cluster export, all services under `global-namespaces` are global by default and customers cannot create global services under non global(local) namespaces.  
```

Comment:
nit: there is still two mentions of `global-namespaces` here

Reply:
Removed it